core/store/models/job_spec.go

```diff
@@ -31,6 +31,27 @@ type InitiatorRequest struct {
 	InitiatorParams `json:"params,omitempty"`
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (i *InitiatorRequest) UnmarshalJSON(buf []byte) error {
```

Comment:
I propose defining the each initiator as their own type and doing something like "encoding and decoding generics" in https://blog.gopheracademy.com/advent-2016/advanced-encoding-decoding/
But I think this would deserve its own PR.